openreview/venue/process/invitation_edit_process.py

```diff
@@ -78,20 +78,36 @@ def update_note_readers(submission, paper_invitation):
         if 'everyone' in invitation_readers and 'everyone' not in submission.readers:
             return
 
+        def updated_content_readers(note, paper_inv):
+            updated_content = {}
+            for key in note.content.keys():
+                invitation_readers = paper_inv.edit['note']['content'].get(key, {}).get('readers', [])
+                if note.content[key].get('readers', []) != invitation_readers:
+                    updated_content[key] = {
+                        'readers': invitation_readers if invitation_readers else { 'delete': True }
+                    }
+            return updated_content
+
         if type(invitation_readers) is list:
             for note in notes:
-                final_invitation_readers = [note.signatures[0] if 'signatures' in r else r for r in invitation_readers]
+                final_invitation_readers = list(set([note.signatures[0] if 'signatures' in r else r for r in invitation_readers]))
```

Comment:
Do you think this is still necessary if we don't allow PCs to sign using the PC group? The order of readers does not match the order in the tests now because of `set()`